auth/proxy/alarms.yaml

```diff
@@ -0,0 +1,242 @@
+AWSTemplateFormatVersion: "2010-09-09"
+
+Description: |
+  SAM template for attestation alarms
+
+Parameters:
+  ApiId:
+    Description: ID of the attestation api
+    Type: String
+
+  ApiLogGroupName:
+    Description: Name of the attestation proxy log group
+    Type: String
+
+  AlarmTopic:
+    Description: The name of Attestation Alerts SNS topic
+    Type: String
+
+  Environment:
+    Description: The name of the environment to deploy to
+    Type: String
+    AllowedValues:
+      - build
+      - staging
+      - production
+      - integration
+      - demo
+      - local
+      - dev
+
+  LambaName:
+    Description: Name of the attestation proxy
+    Type: String
+
+  FunctionLogGroupName:
+    Description: Name of the attestation proxy log group
+    Type: String
+
+Transform: AWS::Serverless-2016-10-31
+
+Resources:
+  Attestation200ResponseCountMetricFilter:
+    Type: AWS::Logs::MetricFilter
+    Properties:
+      FilterPattern: '{ ($.resourcePath = "/oauth2/token") && ($.status = "200") }'
+      LogGroupName: !Ref ApiLogGroupName
+      MetricTransformations:
+        - MetricName: Attestation200ResponseCount
+          MetricNamespace: !Sub ${AWS::StackName}/Responses
+          MetricValue: "1"
+
+  AttestationLambdaErrorRateAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      ActionsEnabled: true
+      AlarmDescription: "Alarm for high number of attestation errors. \nSee resolution steps in runbook: \nhttps://gov-uk.atlassian.net/wiki/spaces/GOVUK/pages/4517658627/Runbook+High+number+of+attestation+failures\n"
+      AlarmName: !Sub ${AWS::StackName}-attestation-lambda-error-rate
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      # The threshold for these alarms can be lower since they are only triggered by
+      # unhandled exceptions so will indicate a serious issue with the lambda code.
+      DatapointsToAlarm: 1
+      EvaluationPeriods: 1
+      Metrics:
```

Comment:
Alerts if the percentage of lambda invocation errors is `>= 10%` per minute for at least `10` invocations. This should give us a quick indication if there any uncaught exceptions. 

Ideally, this should be be part of a canary release - whereby a lambda isn't deployed if this alert is triggered.